lua/nui/layout/utils.lua

```diff
@@ -89,7 +89,7 @@ function mod.get_container_info(position)
 
   return {
     relative = position.bufpos and "buf" or relative,
-    size = utils.get_window_size(position.win),
+    size = utils.get_window_size(relative == "cursor" and 0 or position.win),
```

Comment:
This will just fix the size. But reference to that invalid window id will remain internally.

Can we try this instead?

```diff
diff --git a/lua/nui/layout/utils.lua b/lua/nui/layout/utils.lua
index 6f251b5..66efcc3 100644
--- a/lua/nui/layout/utils.lua
+++ b/lua/nui/layout/utils.lua
@@ -137,7 +137,9 @@ function mod.update_layout_config(component_internal, config)
   if options.relative then
     internal.layout.relative = options.relative
 
-    local fallback_winid = internal.position and internal.position.win or vim.api.nvim_get_current_win()
+    local fallback_winid = internal.position and internal.position.win
+      or internal.layout.relative.type == "cursor" and 0
+      or vim.api.nvim_get_current_win()
     internal.position =
       vim.tbl_extend("force", internal.position or {}, mod.parse_relative(internal.layout.relative, fallback_winid))
 
```

Reply:
Fair enough. I'll use that instead.

What do you think about this?

> [!NOTE]
> It might be better to `relative ~= "win" and 0 or position.win`?
>
> Basically the values are `relative = "editor"|"win"|"cursor"`, and "editor" case is dealt with above, so they mean the exact same thing for now, but may not be in the future (when new position value is added).

Reply:
`nui.nvim` also accepts this: https://github.com/MunifTanjim/nui.nvim/blob/fc59553b5a8a1c13b8aa25ae62b6a47ec2b1882c/lua/nui/popup/README.md#L229-L240

Which internally translates to

```
win = current_window
bufpos = position_relative_to_buffer_on_current_window
```

The `bufpos` is only valid for a specific buffer, so `win` needs to be fixed to a window that contains that buffer.

Reply:
Oh so it is `relative = "editor"|"win"|"cursor"|"buf"`?

Then I'll use `relative == "cursor" and 0`.

Reply:
@MunifTanjim Fixed!